### PR TITLE
Messages: enable modmail filtering by subreddit

### PR DESCRIPTION
**Pull request by curioussavage**
:eyeglasses: @ajacksified

### CODE REVIEW

File: apis/messages.es6.js [hunk 1]
```diff
@@ -31,8 +31,12 @@ export default class Messages extends BaseAPI {
   }
 
   path(method, query={}) {
-    if (method === 'get') { return `message/${query.view || 'inbox'}`; }
-    if (!query.thingId) { return 'api/compose'; }
+    const { subreddit, view, thingId } = query;
```
Comment (ajacksified) [inline apis/messages.es6.js]: Looks good, just one thought- any particular reason this is thingId instead of just id?

Comment (curioussavage) [inline apis/messages.es6.js]: not sure, looks like @nramadas originally did this one.

Comment (curioussavage) [inline apis/messages.es6.js]: Also not sure what it is for, Could that be addressed later? I'm in the trying to get messages from the api into modmail

Comment (ajacksified) [inline apis/messages.es6.js]: Yup, go for it

**Pull request merged**